engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala

```diff
@@ -566,7 +569,7 @@ object PeriodicProcessService {
     def pickMostImportantActiveDeployment: Option[DeploymentStatus] = {
       val lastActiveDeploymentStatusForEachSchedule =
         latestDeploymentForEachSchedule(activeDeploymentsStatuses)
-          .sortBy(_.runAt)(Ordering[LocalDateTime])
+          .sortBy(_.runAtWithCreatedAt)
```

Comment:
it's already sorted. Do we need to sort it again? Or we may ask: how do we know it's sorted? Maybe the sorting inside the `latestDeploymentForEachSchedule` method is not needed if it's hard to express it differently than with the name (it can be done using types, but let's not take this into consideration now). 

So, I see two possible improvements here:
1. get rid of sorting inside the implementation of `latestDeploymentForEachSchedule` and sort in places where the function is used
2. change the name of the method to give a reader the hint that the output collection is sorted (and how). Then in this line, we could just reverse the collection.

IMO the 1st idea is better. WDYT? Maybe another idea?

Reply:
Hmm I am not sure it is fully sorted already. The sort operation inside `latestDeploymentForEachSchedule` is taking place inside `List[List[A]]` and it is taking place to get latest deployment for each schedule and produce `List[A]` which I don't think is sorted yet :thinking: 

As for me I found it quite intuitive to read and I wouldn't change it too much - still I am open to suggestions, I might be wrong :slightly_smiling_face:

Reply:
ok, so let's leave it